packaging/resource_suite_s3_nocache.py

```diff
@@ -2406,6 +2409,67 @@ def test_upload_very_large_file_after_redirect_issues_2104_2114(self):
             s3plugin_lib.remove_if_exists(file1)
             s3plugin_lib.remove_if_exists(retrieved_file)
 
+    def test_iput_with_N_flag__issue_2198(self):
+
+        filename = f'{inspect.currentframe().f_code.co_name}'
+        get_filename = f'{filename}.get'
+        try:
+            file_sizes = [
+                    100,
+                    32 * 1024 * 1024 - 1,
+                    32 * 1024 * 1024,
+                    32 * 1024 * 1024 + 1,
+                    100 * 1024 * 1024
+                    ];
+
+            for file_size in file_sizes:
+                # create the file
+                lib.make_arbitrary_file(filename, file_size)
+
+                # put the file
+                self.user1.assert_icommand(f'iput {filename}')
+                self.user1.assert_icommand(f'iget -f {filename} {get_filename}')
+                assert_command(['diff', '-q', filename, get_filename])
+
+                for n_value in [0, 1, 2,3,4,8]:
+                    self.user1.assert_icommand(f'iput -f -N{n_value} {filename}')
+                    self.user1.assert_icommand(f'iget -f {filename} {get_filename}')
+                    assert_command(['diff', '-q', filename, get_filename])
+
+                # cleanup before next loop
+                s3plugin_lib.remove_if_exists(filename)
+                s3plugin_lib.remove_if_exists(get_filename)
```

Comment:
Consider including these two lines in the `finally` block as well, in the event of a failure in the loop above.

Reply:
done